app/src/main/java/com/github/dedis/student20_pop/PoPApplication.java

```diff
@@ -375,11 +328,48 @@ public void run() {
         });
     }
 
+    /**
+     * Handle received data messages inorder
+     *
+     * @param dataMessages List of received messages
+     */
+    public void handleDataMessages(List<Data> dataMessages) {
+        for(Data data : dataMessages)
+            data.accept(dataHandler);
+    }
+
     /**
      * Type of results when adding a witness
      */
     public enum AddWitnessResult {
         ADD_WITNESS_SUCCESSFUL,
         ADD_WITNESS_ALREADY_EXISTS
     }
+
+    private class PoPDataHandler implements DataHandler {
+
+        @Override
+        public void handle(StateLao stateLao) {
+            Lao lao = laos.get(stateLao.getId());
+            if(lao == null)
+                lao = new Lao(stateLao.getName(),
+                        stateLao.getCreation(),
+                        stateLao.getId(),
+                        stateLao.getOrganizer(),
+                        stateLao.getWitnesses(),
+                        new ArrayList<>(),
+                        new ArrayList<>());
+            else {
+                lao.setName(stateLao.getName());
+                lao.setWitnesses(stateLao.getWitnesses());
+            }
+
+            laos.put(lao.getId(), lao);
+        }
+
+        @Override
+        public void handle(StateMeeting stateMeeting) {
```

Comment:
Same comment as for `handleDataMessages`

Reply:
This is inside an inner class. So it should be at the end.

Reply:
Indeed, didn't see the inner class